lib/publiccloud/azure.pm

```diff
@@ -26,40 +26,20 @@ sub init {
           . $self->key_secret . ' -t ' . $self->tenantid);
 }
 
-sub filename_to_resgroup {
+sub find_img {
     my ($self, $name) = @_;
 
     ($name) = $name =~ m/([^\/]+)$/;
     $name =~ s/\.xz$//;
     $name =~ s/\.vhdfixed$/.vhd/;
-    $name = $self->prefix . "-" . $name;
-    return $name;
-}
-
-sub find_resgroup {
-    my ($self, $name) = @_;
-
-    my $jgroups = decode_json(script_output("az group list"));
-    for my $g (@{$jgroups}) {
-        if ($name eq $g->{name}) {
-            return $name;
-        }
-    }
-
-    return;
-}
-
-sub find_img {
-    my ($self, $name) = @_;
-
-    $name = $self->filename_to_resgroup($name);
-    return unless $self->find_resgroup($name);
-
-    my $images = decode_json(script_output("az image list --resource-group '$name'"));
-
-    #retrives the first image of the resource group
-    return $images->[0]->{name} if (@{$images});
-    return;
+    my $json = script_output("az image show --resource-group openqa-upload --name $name", 60, proceed_on_failure => 1);
+    record_info('INFO', $json);
+    eval {
+        my $image = decode_json($json);
+        return $image->{name};
+    } or do {
```

Comment:
ok, sure.

Reply:
After some investigation, I can't really do that. Look at this code:

    eval {
        my $image = decode_json($json);
        return $image->{name};
    };
    return;

What happens here is that the return in the eval is not executed (or treated as failure by "eval"), so it will always return empty object.

Reply:
Ah yes you right. The return only leave the eval block. I overseen it. And in your previous version it was the last statement of procedure...
If I'm not wrong, then you can just remove that last return and keep the eval as last statement.

Reply:
done